spec/requests/users_spec.rb

```diff
@@ -1,37 +1,185 @@
+      posts_counter: 0
+    )
+  end
+
+  let(:first_post) do
+    Post.create(
+      author: brian,
+      title: 'Welcome to my blog',
+      text: 'This is my first post',
+      likes_counter: 0,
+      comments_counter: 0
+    )
+  end
+
+  let(:second_post) do
+    Post.create(
+      author: brian,
+      title: 'Welcome to my second post',
+      text: 'This is my second post',
+      likes_counter: 0,
+      comments_counter: 0
+    )
+  end
+
+  let(:third_post) do
+    Post.create(
+      author: brian,
+      title: 'Welcome to my third post',
+      text: 'This is my third post',
+      likes_counter: 0,
+      comments_counter: 0
+    )
+  end
+
+  before do
+    tom
+    nancy
+    jane
+    brian
+  end
+
+  context 'index page test group 1' do
+    before { visit root_path }
+
+    it 'shows the username of each user' do
+      expect(page).to have_content(tom.name)
+      expect(page).to have_content(nancy.name)
+      expect(page).to have_content(jane.name)
+      expect(page).to have_content(brian.name)
+    end
+
+    it 'shows the profile picture of each user' do
+      profile_pictures = all('.user-image.border')
+
+      expect(profile_pictures[0]).to be_visible
+      expect(profile_pictures[1]).to be_visible
+      expect(profile_pictures[2]).to be_visible
+      expect(profile_pictures[3]).to be_visible
+      expect(profile_pictures[0]['src']).to eq tom.photo
+      expect(profile_pictures[1]['src']).to eq nancy.photo
+      expect(profile_pictures[2]['src']).to eq jane.photo
+      expect(profile_pictures[3]['src']).to eq brian.photo
+    end
+  end
+
+  context 'index page test group 2' do
+    before { visit root_path }
+
+    it 'shows the number of posts each user has written' do
+      user_tiles = all('.card')
+      user_tiles.each do |user_tile|
+        expect(user_tile).to have_content(/\d+ Post/)
+      end
+    end
+
+    it 'redirects to a user\'s show page on click' do
+      click_link tom.name
+      toms_show_page_url = user_path(id: tom.id)
+      expect(page).to have_current_path(toms_show_page_url)
+    end
+  end
+
+  context 'show page test group 1' do
+    subject { brian }
+    before do
+      first_post
+      second_post
+      third_post
+      visit user_path(subject)
+    end
+
+    it 'shows the user profile picture' do
+      profile_picture = find('.user-image')
+      expect(profile_picture).to be_visible
+      expect(profile_picture['src']).to eq subject.photo
+    end
+
+    it 'shows the user\'s name' do
+      expect(page).to have_content(subject.name)
+    end
+    it 'shows the number of posts the user has written' do
+      expect(page).to have_content("#{subject.posts_counter} Post#{subject.posts_counter == 1 ? '' : 's'}")
+    end
+    it 'shows the user bio' do
+      expect(page).to have_content("Brian Tally\n3 Posts\n
+      Add a post\n@@ -19,19 +19,19 @@\nComments: 0 Likes: 0\nWelcome to my second post")
```

Comment:
- [ ] Please double-check on this test case because it's failing. 🙏 
![pwsh in blogapp 11_15_2023 12_33_04 AM](https://github.com/MarcoDDM/blogapp/assets/47625407/fe0c2557-ce6b-4fd6-bde3-5726868c1363)

Reply:
Ok, thank you, I will check it now...